eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/FluentMethodNameCheck.java

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Model Class Method:
+ *  Fluent Methods: All methods that return an instance of the class, and with one parameter.
+ *  The method name should not start with {@code avoidStartWord}.
+ */
+public class FluentMethodNameCheck extends AbstractCheck {
+
+    private static final String FLUENT_METHOD_ERR = "Fluent Method name should not start with keyword %s.";
+
+    // Specifies valid identifier: default start word is 'with'
+    private String[] avoidStartWords = new String[] {"with"}; // by default
+    private static final String MODEL = ".model";
+    private String className;
+    private static boolean isModelClass;
+
+    @Override
+    public void beginTree(DetailAST ast) {
+        this.isModelClass = false;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.METHOD_DEF
+        };
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.PACKAGE_DEF:
+                FullIdent packageNameFI = FullIdent.createFullIdentBelow(ast);
+                this.isModelClass = packageNameFI.getText().endsWith(MODEL);
+                break;
+            case TokenTypes.CLASS_DEF:
+                if (isModelClass) {
+                    className = ast.findFirstToken(TokenTypes.IDENT).getText();
+                }
+                break;
+            case TokenTypes.METHOD_DEF:
+                if (isModelClass) {
+                    isMethodNameStartWith(ast);
```

Comment:
The method name is misleading because it suggests this method should return a boolean, but it does not.